### PR TITLE
Converted all the code to typescript. …

### PR DESCRIPTION
**Pull request by KolCrooks**
There are still some problems with some of the typings for the parser and the constants have duplicate keys(idk how to fix this). I also don't really know how to test any of this. I would recommend that you merge this into a new branch or clone my fork so that you can make sure that this stuff works. Look at the //TODO comments to see some stuff that I wasn't able to figure out.

### CODE REVIEW

File: dist/solution.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export {};
```
Comment (ANovokmet) [inline dist/solution.d.ts:1]: This declaration it invalid. Solution.ts needs a default export.

File: src/project.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+export default class Project {
+    projectTypeGuid: string;
+    projectName: string;
+    relativePath: string;
+    projectGuid: string;
+    filePath: string;
+    solution?: { path: string; };
+    xmlContent: XMLContent | undefined;
+    //TODO I use any a lot here but I really should have this as a concrete type. I just don't know what type that would be
+    itemGroup: { Reference: any[]; Compile: any[]; None: any[]; Content: any[]} = {Reference: [], Compile: [], None: [], Content: []};
```
Comment (ANovokmet) [inline src/project.ts:33]: This is good. { [key:string]: any[] } is acceptable too.  Remove the comment.

File: src/project.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        this.xmlContent = await parser.parseStringPromise(buffer.toString());
+        
+        if(this.xmlContent)
+        for (const itemGroup of this.xmlContent.Project.ItemGroup) {
+            Object.assign(this.itemGroup, itemGroup);
+        }
+    }
+    
+    //TODO IDK what the property node's type is and it seems like it can really be anything. You would know more about what this can be
+    getPropertyNode(key: string): any {
```
Comment (ANovokmet) [inline src/project.ts:57]: Yeah I know... It's `string`.

File: src/solution.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import fs from "fs-extra";
+import path from "path";
+import SlnParser, { GlobalSectionNode, HeaderNode } from "./parser";
+import Project from "./project";
+
+class Solution {
```
Comment (ANovokmet) [inline src/solution.ts]: export default

**Review by ANovokmet — COMMENTED** [review]
Nice work. There is one thing missing, index.js should be rewritten into typescript too. You can move it to /src. Also, when using require for default export we use `Solution: require('./dist/solution').default`.

**Comment by KolCrooks** [comment]
I added the changes that you mentioned. I also changed the parser to have different files. I can revert these changes if you want. The only downside to the way I changed it is that when you import the parser, you have to do `import * as SlnParser` instead of `import SlnParser`. Other than that I think that this is pretty good.

**Comment by ANovokmet** [comment]
Yes, it looks pretty good. Thanks!

**Pull request merged**